src/libxtp/md2qmengine.cc

```diff
@@ -143,27 +143,27 @@ Topology Md2QmEngine::map(const csg::Topology& top) {
   }
 
   int atomid = 0;
-  for (const csg::Molecule* mol : top.Molecules()) {
-    const std::vector<int> resnums_map = MolToResNum[mol->getName()];
-    const std::vector<std::string> segnames = SegsinMol[mol->getName()];
+  for (const csg::Molecule& mol : top) {
+    const std::vector<int> resnums_map = MolToResNum[mol.getType()];
+    const std::vector<std::string> segnames = SegsinMol[mol.getType()];
 
     std::vector<Segment*> segments;  // we first add them to topology and then
                                      // modify them via pointers;
     for (const std::string& segname : segnames) {
       segments.push_back(&xtptop.AddSegment(segname));
-      segments.back()->AddMoleculeId(mol->getId());
+      segments.back()->AddMoleculeId(mol.getId());
     }
 
     // we have to figure out how the Residue numbers change from molecule to
     // molecule to get the correct mapping information this does not require for
     // the atoms to be sorted according to resnum
     int ResNumOffset = DetermineResNumOffset(mol, resnums_map);
-    for (const csg::Bead* bead : mol->Beads()) {
+    for (const csg::Bead* bead : mol.getBeads()) {
       Segment* seg =
-          segments[MolToSegMap[mol->getName()][bead->getResnr() - ResNumOffset]
-                              [bead->getName()]];
+          segments[MolToSegMap[mol.getType()][bead->getResidueId() - ResNumOffset]
```

Comment:
what is the difference between type and name?

Reply:
Originally, the csg bead had three attributes associated with the name member, they were stored in a single string name_ with the following format:

"beadtype::redue_type::resiude_number"

This I thought was redundant and confusing, because the beads also had separate members for the residue number, so I split the pieces up into their own parts. I.e. the residue number is stored as an integer only and not in a name member, the residue type as a string and the bead type as a separate string. To get the combination one can call getLabel  this will append the different components together as a string. To be consistent with this approach the molecule method getName was also changed to getType. I don't know if I'm being clear or not...